app/src/main/java/org/listenbrainz/android/model/feed/FeedEventType.kt

```diff
@@ -183,22 +186,38 @@ enum class FeedEventType (
                 goToUserPage = goToUserPage,
                 goToArtistPage = goToArtistPage
             )
-            LISTEN -> ListenFeedLayout(
-                event = event,
-                parentUser = parentUser,
-                onDeleteOrHide = onDeleteOrHide,
-                onDropdownClick = onDropdownClick,
-                onClick = onClick,
-                dropdownState = dropDownState,
-                index = index,
-                onOpenInMusicBrainz = onOpenInMusicBrainz,
-                onPin = onPin,
-                onReview = onReview,
-                onPersonallyRecommend = onPersonallyRecommend,
-                onRecommend = onRecommend,
-                goToUserPage = goToUserPage,
-                goToArtistPage = goToArtistPage
-            )
+            LISTEN -> {
+                val listensViewModel: ListensViewModel = hiltViewModel()
```

Comment:
Rather than creating a new view-model instance here, `onDeleteListen` should be passed in as a callback.